xarray/plot/plot.py

```diff
@@ -433,11 +470,16 @@ def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
         # This allows the user to pass in a custom norm coming via kwargs
         kwargs.setdefault('norm', cmap_params['norm'])
 
-        ax, primitive = plotfunc(xval, yval, zval, ax=ax,
-                                 cmap=cmap_params['cmap'],
-                                 vmin=cmap_params['vmin'],
-                                 vmax=cmap_params['vmax'],
-                                 **kwargs)
+        if 'imshow' == plotfunc.__name__ and isinstance(aspect, basestring):
```

Comment:
Yes, sorry. Both methods will raise the same ``ValueError`` if called with ``aspect=2``:

```
ValueError: cannot provide `aspect` argument without `size`
```
The point of this line is to provide a more informative error message when users try to set aspect='equal', as in [this test](https://github.com/pydata/xarray/blob/master/xarray/test/test_plot.py#L901) for example. If you're happy with the message above I can remove this block.